packages/fiber/src/web/Canvas.tsx

```diff
@@ -59,27 +62,36 @@ export function Canvas({ children, fallback, tabIndex, resize, id, style, classN
         <ErrorBoundary set={setError}>
           <React.Suspense fallback={<Block set={setBlock} />}>{children}</React.Suspense>
         </ErrorBoundary>,
-        canvas.current,
+        canvasRef.current,
         { ...props, size, events: events || createPointerEvents },
       )
     }
   }, [size, children])
 
   useIsomorphicLayoutEffect(() => {
-    const container = canvas.current
+    const container = canvasRef.current!
     return () => unmountComponentAtNode(container)
   }, [])
 
   return (
     <div
-      ref={ref}
+      ref={containerRef}
       id={id}
       className={className}
       tabIndex={tabIndex}
       style={{ position: 'relative', width: '100%', height: '100%', overflow: 'hidden', ...style }}>
-      <canvas ref={canvas} style={{ display: 'block' }}>
+      <canvas
+        ref={(node) => {
+          ;(canvasRef as React.MutableRefObject<HTMLCanvasElement | null>).current = node
```

Comment:
i'd prefer https://github.com/gregberge/react-merge-refs

```jsx
<canvas ref={merge([canvasRef, ref])}
```

we use this in drei as well